src/util/request-manager.js

```diff
@@ -389,15 +417,25 @@ export default class RequestManager {
 
         this.reporter.verbose(this.reporter.lang('verboseRequestFinish', params.url, res.statusCode));
 
+        if (res.statusCode === 408 || res.statusCode >= 500) {
+          const description = `${res.statusCode} ${http.STATUS_CODES[res.statusCode]}`;
+          if (!queueForRetry(this.reporter.lang('internalServerErrorRetrying', description))) {
+            throw new ResponseError(this.reporter.lang('requestFailed', description), res.statusCode);
+          } else {
+            return;
+          }
+        }
+
         if (body && typeof body.error === 'string') {
           reject(new Error(body.error));
           return;
         }
 
-        if (res.statusCode === 403) {
+        if ([403, 408].indexOf() !== -1 || res.statusCode >= 500) {
```

Comment:
Isn't `res.statusCode` mising in the `indexOf()`?

Reply:
Yep, it totally does ... Flow should have warned me 😞